[WebKit CI] Double the timeout for site isolation layout testing queues.

bls1999 opened the pull request:
#### 679c35a0655fcfa076613b8683f9eaec4334382d
<pre>
[WebKit CI] Double the timeout for site isolation layout testing queues.
<a href="https://bugs.webkit.org/show_bug.cgi?id=290863">https://bugs.webkit.org/show_bug.cgi?id=290863</a>
<a href="https://rdar.apple.com/148365603">rdar://148365603</a>

Reviewed by Brianna Fan.

Layout testing in site isolation mode takes longer than regular layout testing.
Therefore, we&apos;d like to double the timeout threshold from 3 -&gt; 6 hours for
site isolation layout testing specifically.

* Tools/CISupport/build-webkit-org/steps.py:
(RunWebKitTests.run): Double the layout-tests step timeout, change from 3 -&gt; 6 hours.

Canonical link: <a href="https://commits.webkit.org/293039@main">https://commits.webkit.org/293039@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d08b8c801001978041a23c4c86a69b197f71f0c9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97796 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17421 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7638 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/102908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/48325 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/99841 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/17715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/25874 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/102908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/48325 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100799 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/17715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/102908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/17715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/6305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/47768 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/17715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Tests-EWS "Waiting to run tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/105288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/24876 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/25874 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/105288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/97366 "Passed tests") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/25248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/84553 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/105288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/18483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15812 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/24837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/24659 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/27973 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/26233 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -911,6 +911,10 @@ def run(self):
+            # Double the timeout for site isolation queues.
+            # TODO: Expose this property in config.json.
+            if '--site-isolation' in additionalArguments:
+                self.timeout = 6 * 60 * 60
```
Can we verify this behaves as expected through a unit test? You can see existing ones in `TestRunWebKitTests`!

briannafan commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -911,6 +911,10 @@ def run(self):
         if additionalArguments:
             self.command += additionalArguments
+            # Double the timeout for site isolation queues.
+            # TODO: Expose this property in config.json.
```
Could you specify which property? Timeout or site isolation?
And nit: Someone once told me to use `# FIXME` instead of `# TODO`.

JonWBedard replied:
Probably "expose this _as a_ property in config.json

bls1999 replied:
Good point, I'll change `TODO` -> `FIXME`.

~Can you elaborate on what "specify which property" means in this context?~ ah gotcha, which property to expose.

briannafan commented on Tools/CISupport/build-webkit-org/steps_unittest.py:
```diff
@@ -958,6 +958,42 @@ def test_wpe_parameters(self):
         self.expectOutcome(result=SUCCESS, state_string='layout-tests')
         return self.runStep()
 
+    def test_default_timeout(self):
```
nit: The other tests already cover the default timeout, probably can delete this and keep `test_site_isolation_timeout`.

bls1999 replied:
Great point, I was actually just making this change lol

briannafan commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -911,6 +911,10 @@ def run(self):
         if additionalArguments:
             self.command += additionalArguments
+            # Double the timeout for site isolation queues.
+            # FIXME: Expose the timeout property in config.json.
```
What would actually be the better fix is https://bugs.webkit.org/show_bug.cgi?id=290867, so we don't need to rely on setting huge timeouts for each individual test. Feel free to include this in the `FIXME` but r+ nonetheless!

bls1999 committed:
[WebKit CI] Double the timeout for site isolation layout testing queues.
https://bugs.webkit.org/show_bug.cgi?id=290863
rdar://148365603

Reviewed by Brianna Fan.

Layout testing in site isolation mode takes longer than regular layout testing.
Therefore, we'd like to double the timeout threshold from 3 -> 6 hours for
site isolation layout testing specifically.

* Tools/CISupport/build-webkit-org/steps.py:
(RunWebKitTests.run): Double the layout-tests step timeout, change from 3 -> 6 hours.

Canonical link: https://commits.webkit.org/293039@main
